oauth2_http/java/com/google/auth/http/HttpCredentialsAdapter.java

```diff
@@ -44,16 +55,58 @@ public void initialize(HttpRequest request) throws IOException {
     }
     for (Map.Entry<String, List<String>> entry : credentialHeaders.entrySet()) {
       String headerName = entry.getKey();
-      List<String> requestValues = getHeadersValue(requestHeaders, headerName);
-      if (requestValues == null) {
-        requestValues = new ArrayList<String>();
-        requestHeaders.put(headerName, requestValues);
+      List<String> requestValues = new ArrayList<String>();
+      requestValues.addAll(entry.getValue());
+      requestHeaders.put(headerName, requestValues);
+    }
+  }
+
+  /**
+   * {@inheritDoc}
+   * <p>
+   * Checks if {@code WWW-Authenticate} exists and contains a "Bearer" value
+   * (see <a href="http://tools.ietf.org/html/rfc6750#section-3.1">rfc6750 section 3.1</a> for more
+   * details). If so, it refreshes the token in case the error code contains
+   * {@code invalid_token}. If there is no "Bearer" in {@code WWW-Authenticate} and the status code
+   * is {@link HttpStatusCodes#STATUS_CODE_UNAUTHORIZED} it refreshes the token. If
+   * the token refresh throws an I/O exception, this implementation will log the
+   * exception and return {@code false}.
+   * </p>
+   * NOTE: mostly copied from Credential.java in google-oauth-java-client
```

Comment:
I'd be tempted to leave this out. It is semi-true for a lot of code here. May be better to mention in a central readme.